packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkNativeInitializationTest.kt

```diff
@@ -0,0 +1,168 @@
+            assertThat(it.address.hostAddress).isEqualTo("1.1.1.1")
+        }
+        assertThat(configuration.proxyConfig?.second?.username).isEqualTo("proxyusername")
+        assertThat(configuration.proxyConfig?.second?.password).isEqualTo("proxypassword")
+        assertThat(configuration.firstPartyHosts?.get("example.com").toString()).isEqualTo(
+            "[B3MULTI, TRACECONTEXT]"
+        )
+    }
+
+    @Test
+    fun `𝕄 add defaults to config 𝕎 getConfigurationFromJSONFile() {file with minimal data}`() {
+        whenever(
+            mockJSONFileReader.parseAssetsJSONFile(mockContext, "datadog-configuration.json")
+        ) doReturn javaClass.getResource(
+            "/input/minimal-configuration.json"
+        ).readText()
+        val configuration = testedNativeInitialization.getConfigurationFromJSONFile()
+        assertThat(configuration.clientToken).isEqualTo("fake-client-token")
+        assertThat(configuration.env).isEqualTo("fake-env")
+        assertThat(configuration.applicationId).isEqualTo("fake-app-id")
+        assertThat(configuration.nativeCrashReportEnabled).isEqualTo(false)
+        assertThat(configuration.nativeLongTaskThresholdMs).isEqualTo(200.0)
+        assertThat(configuration.longTaskThresholdMs).isEqualTo(0.0)
+        assertThat(configuration.sampleRate).isEqualTo(100.0)
+        assertThat(configuration.site).isEqualTo("US1")
+        assertThat(configuration.trackingConsent).isEqualTo("GRANTED")
+        assertThat(configuration.telemetrySampleRate).isEqualTo(20.0)
+        assertThat(configuration.vitalsUpdateFrequency).isEqualTo("AVERAGE")
+        assertThat(configuration.trackFrustrations).isEqualTo(true)
+        assertThat(configuration.uploadFrequency).isEqualTo("AVERAGE")
+        assertThat(configuration.batchSize).isEqualTo("MEDIUM")
+        assertThat(configuration.trackBackgroundEvents).isEqualTo(false)
+        assertThat(configuration.customEndpoints).isNull()
+        assertThat(configuration.additionalConfig?.get("_dd.source")).isEqualTo("react-native")
+        assertThat(configuration.additionalConfig?.get("_dd.sdk_version")).isEqualTo(SDK_VERSION)
+        assertThat(configuration.configurationForTelemetry).isNull()
+        assertThat(configuration.nativeViewTracking).isEqualTo(false)
+        assertThat(configuration.nativeInteractionTracking).isEqualTo(false)
+        assertThat(configuration.verbosity).isNull()
+        assertThat(configuration.serviceName).isNull()
+        assertThat(configuration.proxyConfig).isNull()
+        assertThat(configuration.firstPartyHosts).isNull()
+    }
+
+    @Test
+    fun `𝕄 throw error 𝕎 getConfigurationFromJSONFile() {incorrect file}`() {
+        whenever(
+            mockJSONFileReader.parseAssetsJSONFile(mockContext, "datadog-configuration.json")
+        ) doReturn javaClass.getResource(
+            "/input/malformed-configuration.json"
+        ).readText()
+        assertThrows<NullPointerException> {
```

Comment:
should it be some specific exception instead? Some `BadConfigurationException` or something like that? `NullPointerException` is kind of too generic for that.